v2/dtos/requests/device.go

```diff
@@ -0,0 +1,167 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package requests
+
+import (
+	"encoding/json"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+)
+
+// AddDeviceRequest defines the Request Content for POST Device DTO.
+// This object and its properties correspond to the AddDeviceRequest object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/AddDeviceRequest
+type AddDeviceRequest struct {
+	common.BaseRequest `json:",inline"`
+	DeviceName         string                             `json:"deviceName" validate:"required"`
+	ServiceName        string                             `json:"serviceName" validate:"required"`
+	ProfileName        string                             `json:"profileName" validate:"required"`
+	AdminState         string                             `json:"adminState"`
+	OperatingState     string                             `json:"operatingState"`
+	AutoEvents         []dtos.AutoEvent                   `json:"autoEvents"`
+	Protocols          map[string]dtos.ProtocolProperties `json:"protocols" validate:"required,gt=0"`
+}
+
+// Validate satisfies the Validator interface
+func (d AddDeviceRequest) Validate() error {
+	err := v2.Validate(d)
```

Comment:
add logic to validate AutoEvent.Frequency format

Reply:
Added.